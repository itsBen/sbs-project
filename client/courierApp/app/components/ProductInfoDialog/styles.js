import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    borderRadius: 5,
    borderColor: 'rgba(0, 0, 0, 0.1)',
  },
  headerContainer: {
    backgroundColor: 'rgb(230,37,101)',
    flexDirection: 'row',
    borderTopLeftRadius: 5,
    borderTopRightRadius: 5,
    position: 'relative',
    minHeight: 50,
  },
  header: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  headerText: {
    fontSize: 22,
    fontWeight: 'bold',
    paddingVertical: 10,
    color: '#fff',
  },
  body: {
    flex: 1,
    flexDirection: 'row',
    paddingHorizontal: 22,
  },
  bodyBody: {
    flex: 1,
  },
  footer: {
    flexDirection: 'row',
    paddingHorizontal: 22,
    paddingBottom: 22,
    position: 'absolute',
    bottom: 0,
  },
  footerBody: {
    flex: 1,
  },
  imageContainer: {
    alignItems: 'center',
    paddingVertical: 20,
  },
  image: {
    width: 120,
    height: 120,
    borderRadius: 5,
  },
  pricelabel: {
    fontSize: 19,
    fontWeight: 'bold',
    textAlign: 'center',
    marginTop: 10,
    marginBottom: 20,
  },
  metatable: {
    borderWidth: 1,
    borderColor: '#e0e0e0',
    borderRadius: 2,
    marginBottom: 20,
  },
  metatableElement: {
    borderBottomWidth: 1,
    borderBottomColor: '#e0e0e0',
    paddingTop: 5,
    paddingBottom: 5,
    paddingLeft: 20,
    paddingRight: 20,
  },
  metatableElementLabel: {
    color: '#333333',
    marginBottom: 3,
  },
  metatableElementValue: {
    color: '#4D4D4D',
  },
  spinner: {
    justifyContent: 'center',
    alignSelf: 'center',
    paddingTop: 70,
  },
});
