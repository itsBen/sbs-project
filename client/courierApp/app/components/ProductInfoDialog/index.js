import React, { Component } from 'react';
import { Image, StyleSheet, Text, View, ActivityIndicator } from 'react-native';
import API from '@api';

import { SmallButton } from '../Button';
import styles from './styles';

export default class extends Component {
  state = {
    product: null,
  };

  componentWillMount() {
    API.getProductById(this.props.productId).then(res => {
      if (res.items.length > 0) {
        console.log(res.items[0].fields);
        this.setState({ product: res.items[0].fields });
      } else {
        window.alert('Error!');
      }
    });
  }

  render() {
    const { product } = this.state;
    return (
      <View style={styles.container}>
        <View style={styles.headerContainer}>
          <View style={styles.header}>
            <Text style={styles.headerText}>{product ? product.name : ''}</Text>
          </View>
        </View>
        {!product ? (
          <ActivityIndicator size="large" style={styles.spinner} />
        ) : (
          <View style={styles.body}>
            <View style={styles.bodyBody}>
              <View style={styles.imageContainer}>
                <Image
                  style={styles.image}
                  source={{
                    uri: product.photoURL
                      ? product.photoURL
                      : 'https://facebook.github.io/react/img/logo_og.png',
                  }}
                />
              </View>

              <Text style={styles.pricelabel}>{product.price} â‚¬</Text>

              <View style={styles.metatable}>
                <View style={styles.metatableElement}>
                  <Text style={styles.metatableElementLabel}>Size</Text>
                  <Text style={styles.metatableElementValue}>
                    {product.size &&
                      product.size.value + ' ' + product.size.unit}
                  </Text>
                </View>
                <View style={styles.metatableElement}>
                  <Text style={styles.metatableElementLabel}>Brand</Text>
                  <Text style={styles.metatableElementValue}>
                    {product.brand}
                  </Text>
                </View>
                <View style={styles.metatableElement}>
                  <Text style={styles.metatableElementLabel}>Dimensions</Text>
                  <Text style={styles.metatableElementValue}>
                    {product.dimensions}
                  </Text>
                </View>
              </View>
            </View>
          </View>
        )}
        <View style={styles.footer}>
          <View style={styles.footerBody}>
            <View style={{ height: 10 }} />
            <SmallButton
              title="Cancel"
              buttonStyle="white"
              onPress={this.props.onClose}
            />
          </View>
        </View>
      </View>
    );
  }
}
