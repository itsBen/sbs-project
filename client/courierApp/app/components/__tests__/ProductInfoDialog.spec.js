/* eslint-env node, jest */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ProductInfoDialog from '../ProductInfoDialog';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <ProductInfoDialog
        product={{
          name: 'Meat',
          size: {
            unit: 'l',
            value: 2.6,
          },
          dimensions: 'small',
          price: 23.4,
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
