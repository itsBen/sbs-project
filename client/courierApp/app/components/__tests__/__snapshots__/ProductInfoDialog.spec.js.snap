// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "rgba(0, 0, 0, 0.1)",
      "borderRadius": 5,
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgb(230,37,101)",
        "borderTopLeftRadius": 5,
        "borderTopRightRadius": 5,
        "flexDirection": "row",
        "minHeight": 50,
        "position": "relative",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "fontSize": 22,
            "fontWeight": "bold",
            "paddingVertical": 10,
          }
        }
      >
        
      </Text>
    </View>
  </View>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "alignSelf": "center",
        "justifyContent": "center",
        "paddingTop": 70,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "flexDirection": "row",
        "paddingBottom": 22,
        "paddingHorizontal": 22,
        "position": "absolute",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "height": 10,
          }
        }
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "rgb(230,37,101)",
                "borderColor": "rgb(230,37,101)",
                "borderRadius": 5,
                "borderWidth": 1,
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              },
              Object {
                "backgroundColor": "#fff",
                "borderColor": "transparent",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#FFF",
                  "fontSize": 17,
                },
                Object {
                  "color": "#808080",
                },
              ]
            }
          >
            Cancel
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
